separate rules for whether a tab can be created discarded and whether it can be autoDiscardable

petemill opened the pull request:
Fix #12976

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

Test Plan:


Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests


- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

bridiver commented on app/browser/tabs.js:
```diff
@@ -912,8 +912,16 @@ const api = {
+      if (createProperties.pinned || !isHttpOrHttps(createProperties.url)) {
         createProperties.discarded = false
+      }
+      // similarly, autoDiscardable will happen for regular tabs (not about: tabs)
```
should have a note that this is temporarily disabled and link to the issue

petemill replied:
it's still in the separate function, perhaps I should just put that logic inline here too since there's only 1 usage and it's 1 line. thoughts?

bridiver replied:
That seems fine. This is the problem with these github reviews sometimes because important code is often not visible. The original issue with `isAutoDiscardable` was missed for a similar reason

petemill replied:
@bridiver I inlined it for now. Normally I like the functional / semantic language of bundling that logic out, but it was causing confusion in this case.

bridiver commented on app/browser/tabs.js:
```diff
@@ -112,12 +112,6 @@ const needsPartitionAssigned = (createProperties) => {
 }
 
-const isAutoDiscardable = (createProperties) => {
-  // Temporarily disabled due to bad handling for auto discarded tabs
```
I know we can get it from the history and I should have brought this up with the original PR, but I think we should keep this method and preserve the original logic, but comment it out for now and replace with `false`

petemill commented:
Addressed @bridiver feedback and made a separation in defined logic between wheter a tab is 'auto-discardable' and whether a tab can be 'lazy-loaded'. Also moved logic inline to `tabs.create:` fn since this module is very large and is maybe more readable to put this all in one place to avoid anymore future confusion.

petemill commented:
master 8314998
0.22.x 82877c228823cfa85516bf93ce8152eeb7585dd4
0.21.x 9dc1f00cc6fe59486f13a322e934a47d67652f76
0.20.x d53e4381a0dd9d62565ba07192c3f893104b226a
